client/my-sites/controller.js

```diff
@@ -109,6 +111,43 @@ function renderNoVisibleSites( context ) {
 	);
 }
 
+function isPathAllowedForDomainOnlySite( pathname ) {
+	const urlPrefixesWhiteListForDomainOnlySite = [
+		'/domains/manage/',
+		'/checkout/',
+	];
+
+	return urlPrefixesWhiteListForDomainOnlySite.some( path => startsWith( pathname, path ) );
+}
+
+function onSelectedSiteAvailable( context ) {
+	const selectedSite = sites.getSelectedSite();
+	siteStatsStickyTabActions.saveFilterAndSlug( false, selectedSite.slug );
+	context.store.dispatch( receiveSite( selectedSite ) );
+	context.store.dispatch( setSelectedSiteId( selectedSite.ID ) );
+
+	const state = context.store.getState();
+
+	if ( isDomainOnlySite( state, selectedSite.ID )
+		&& ! isPathAllowedForDomainOnlySite( context.pathname ) ) {
```

Comment:
Also, super minor, but the linter complains unless && is at the end of a line when there are line breaks mid-condition. :)